extension/basicauth/authdata.go

```diff
@@ -0,0 +1,77 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package basicauth // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/basicauth"
+
+import (
+	"encoding/base64"
+	"errors"
+	"strings"
+
+	"go.opentelemetry.io/collector/client"
+)
+
+var (
+	errInvalidSchemePrefix = errors.New("invalid authorization scheme prefix")
+	errInvalidFormat       = errors.New("invalid format")
+)
+
+// See: https://github.com/golang/go/blob/1a8b4e05b1ff7a52c6d40fad73bcad612168d094/src/net/http/request.go#L950
+func parseBasicAuth(auth string) (*authData, error) {
```

Comment:
I feel this function would be more appropriate within `basicauth.go` - it is the actual authentication business logic, not just a data structure (this file is called `authdata.go`)